### DISCUSSION TITLE
Datenpunkt Formatierung mit Hilfe von Variablen

### DISCUSSION THREAD

**Discussion by KaJu99** [Zefau/ioBroker.jarvis#1593] [Q&A]
Habe jetzt schon eine ganze Zeit rumprobiert, komme aber nicht weiter; wie lässt sich (als JSON-Code) die Darstellung z.B. von Datenpunkten in Abhängigkeiten von Werten, die nicht statisch sind, umsetzen. Z.B. Darstellung der IST-Temperatur eines Heizungsventils rot, wenn die Ist-Temperatur kleiner als die aktuelle Solltemperatur ist. Oder Darstellung einer Gerätebeschreibung (zum Beispiel Staus eines Fensters) rot, wenn die Batterie den Zustand "niedrig" meldet.

**Comment by Zefau** [comment]
Siehe https://mcuiobroker.gitbook.io/jarvis-infos/jarvis/v3-fuer-einsteiger/konfigurations-tabs/geraete/geraete-dp#icon-stil

Statt `false` dann `<20` oder so.

**Comment by Zefau** [comment]
@mcuiobroker hast du ein gutes Beispiel in dem Wiki mit Operatoren `<` bzw. `>`?

**Reply by mcuiobroker** [reply]
Ich schau mal, sonst erstelle ich eins.
Ist mehr Thema Binding.
https://mcuiobroker.gitbook.io/jarvis-infos/jarvis/besonderheiten-v3/geraet/binding

**Comment by mcuiobroker** [comment]
@KaJu99 Bitte die DPs und deren Abhängigkeiten und was genau soll sich ändern, dann können wir gemeinsam eine Lösung finden.

**Reply by KaJu99** [reply]
z.B. bei dem DP Ist-Temperatur ("temperature") eines HM Heizkörperthermostats die farbliche Darstellung der Temperatur (Datenpunkt Stil) in Abhängigkeit formatieren (rot/blau) ob die Ist-Temperatur größer (blau) oder kleiner (rot) als die Soll-Temperatur (die liefert ein anderer Datenpunkt "setTemperature") ist. Mit statischen Werten ist das ja kein Thema, hier müsste aber eine Variable oder direkt der andere DP als Bezugswert (im Sinne "<setTemperatur":{"color ...    ) etc. Auf diese Weise liese sich auch der Icon-Stil z.B. durch irgenwelche Werte wie die Batteriespannung formatieren.

**Reply by mcuiobroker** [reply]
Ist ja in dem Beispiel vom Binding gezeigt.
```
{"{Deine setTemperature-ID}<20":{"color":....}"}
```

**Reply by KaJu99** [reply]
Danke; habe jetzt gemerkt, dass dazu wohl die PRO-Version nötig ist; leider funktioniert aktuell die Anmeldung/Kauf via Jarvis nicht. Werde es daher später probieren.

**Reply by KaJu99** [reply]
Ergänzung: um Bindings zu nutzen ist die PRO-Version nötig! Dann funktioniert das wie gewünscht.
